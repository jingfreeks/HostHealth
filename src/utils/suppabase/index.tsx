export {default as supabase} from './suppabase';
