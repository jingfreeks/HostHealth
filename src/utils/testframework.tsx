import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
