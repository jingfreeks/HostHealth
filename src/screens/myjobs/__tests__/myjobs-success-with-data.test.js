import React from 'react';
import {fireEvent,act} from '@testing-library/react-native';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import {renderWithProviders} from '@/utils/testframeworknew';
import {waitFor} from '@testing-library/react-native';
import Myjobs from '../myjobs';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);
jest.mock('@supabase/supabase-js');
jest.useFakeTimers();
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      navigate: () => {},
    }),
    useIsFocused: () => true,
    useDispatch: () => ({dispatch: jest.fn()}),
  };
});

jest.mock('@/slice/myjobs',()=>({
  useGetMyJobsQuery:()=>{
    return{
      isSuccess:true,
      refetch:jest.fn(),
      isLoading:false,
      data:{ids:['1233323333','23444232323']},
      myjobs: {
        __v: 0,
        _id: '1233323333',
        address: 'Metro Manila',
        name: 'Bank of Philippine Island',
      }
    }
  }
}))
describe('My Jobs Screen', () => {
  
  it('Should work as expected to get snapshot', () => {

    const all = renderWithProviders(<Myjobs />,{
      preLoadedState:{auth:{userId:'2331222222'}}
    });
    waitFor(() => {
      expect(all.toJSON()).toMatchSnapshot();
    });
  });
  it('Should work to trigger on flatlist on Refresh function', async() => {

    const all = renderWithProviders(<Myjobs />);
    const el = all.getByTestId('MyJobsListingTestId')
    const {refreshControl} = el.props;
    await act(async () => {
      refreshControl.props.onRefresh();
    });
    // fireEvent(el,onRefresh)
    // waitFor(() => {
    //   expect(all.toJSON()).toBeTruthy();
    // });
  });
});
