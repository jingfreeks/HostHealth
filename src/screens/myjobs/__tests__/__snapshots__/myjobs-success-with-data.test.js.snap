// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Jobs Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    accessibilityLabel="MyJobsListingTestId"
    data={
      [
        "1233323333",
        "23444232323",
      ]
    }
    extraData={
      [
        "1233323333",
        "23444232323",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      [
        {},
      ]
    }
    testID="MyJobsListingTestId"
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
